src/poll-until-promise.ts

```diff
@@ -123,19 +131,26 @@ export class PollUntil {
   }
 
   _shouldStopTrying() {
-    return this._timeFromStart() > this._timeout;
+    return this._timeFromStart() > this._timeout || this._attemptsExhausted();
+  }
+
+  _attemptsExhausted() {
+    return this._maxAttempts !== undefined && this._attemptCount > this._maxAttempts;
   }
 
   _executeAgain() {
     this._log('executing again');
     const currentInterval = this._interval;
     const nextInterval = currentInterval * this._backoffFactor;
     this._interval = (nextInterval > this._backoffMaxInterval) ? this._backoffMaxInterval : nextInterval;
+    this._attemptCount += 1;
     setTimeout(this._runFunction.bind(this), currentInterval);
   }
 
   _failedToWait() {
-    let waitErrorText = `${ERRORS.FAILED_TO_WAIT} after ${this._timeFromStart()}ms`;
+    let waitErrorText = this._attemptsExhausted()
+      ? `Operation unsuccessful after ${this._maxAttempts} attempts`
```

Comment:
maybe we should add the timed passed 
```suggestion
      ? `Operation unsuccessful after ${this._maxAttempts} attempts (${this._timeFromStart()}ms)`
```

Reply:
Done. Tho for this and next comment I standardized the format to be `<original message> (total of X)` where X is, for example, "25ms" or "3 attempts" depending on the scenario.